draft-foudil-securitytxt.md

```diff
@@ -453,26 +466,105 @@ unstructured     = <imported from section 3.2.5 of [RFC5322]>
 
 "ext-field" refers to extension fields, which are discussed in {{extensibility}}
 
-# Security considerations
+# Security Considerations
+
+## Compromised Files and Redirects
+
+An attacker that has compromised a website is able to compromise
+the "security.txt" file as well or setup a redirect to their own site.
+This can result in security reports not being received by the organization
+or sent to the attacker.
+
+To protect against this, organizations SHOULD digitally sign their "security.txt"
+files (as per {{signature}}), use the canonical directive to sign the location
+of the file (as per {{canonical}}), and regularly monitor the file and
+the referenced resources to detect tampering.
+
+Security researchers SHOULD check verify the "security.txt" including the
+signature and checking historical records before using the information. If
+the information in a "security.txt" file looks suspicious or compromised,
+it SHOULD NOT be used.
+
+To avoid redirect attacks, redirects for these files MUST NOT be followed
+if they lead to a different domain (as per {{weblocation}}).
+
+## Incorrect or Stale Information
+
+If information and resources referenced in a "security.txt" file are incorrect
+or not kept up to date, this can result in security reports not being received
+by the organization or sent to incorrect contacts, thus exposing possible
+security issues to third parties.
+
+Organizations SHOULD ensure that any this file and any resources such as web pages,
+email addresses and telephone numbers references by a "security.txt"
+file are kept current, are accessible, controlled by the organization,
+and are kept secure.
+
+## Intentionally Malformed Files, Resources and Reports
+
+It is possible for attackers to generate files that are extraordinarily
+large or otherwise malformed in an attempt to discover or exploit weaknesses
+in parsing code. Implementors SHOULD make sure that any such code
+be robust against large and malformed files. ABNF grammar (as defined in
+{{abnf}}) SHOULD be used a way to verify these files.
+
+Same concerns apply to any other resources referenced within security.txt
+files, as well as any security reports received as the result of publishing
+this file. Such resources and reports may be hostile, malformed or malicious.
+
+## No Implied Permission for Testing
+
+The presence of a security.txt file can be interpreted as providing permission
+to do security testing against that asset. This can lead to increased testing
+against an organization by researchers. On the other hand, a decision not
+to publish a security.txt file can be interpreted by an organization as
+signaling to researchers that permission to test is denied.
+This can lead to pushback against researchers reporting security issues
+to organization.
+
+Therefore implementors MUST NOT assume that presence or absence
+of a security.txt files grants or denies permission for security testing.
```

Comment:
"a security.txt files" -- consider making "files" singular to match with the article:

```suggestion
of a security.txt file grants or denies permission for security testing.
```